docker-entrypoint.sh

```
@@ -0,0 +1,26 @@
+  echo "Received SIGTERM signal, shutting down gracefully..."
+  # Forward the signal to the Node.js process
+  kill -SIGTERM "$child"
+  
+  # Wait for the process to terminate
+  wait "$child"
+  echo "Application terminated gracefully"
+  exit 0
+}
+
+# Set up the trap for SIGTERM
+trap term_handler SIGTERM
```

Shellcheck (0.10.0)

[warning] In POSIX sh, prefixing signal names with 'SIG' is undefined.

(SC3048)